modules/apps/27-interchain-accounts/host/keeper/keeper.go

```diff
@@ -106,6 +109,22 @@ func (k Keeper) GetAppVersion(ctx sdk.Context, portID, channelID string) (string
 	return k.ics4Wrapper.GetAppVersion(ctx, portID, channelID)
 }
 
+// GetAppMetadata retrieves the interchain accounts metadata from the store associated with the provided portID and channelID
+func (k Keeper) GetAppMetadata(ctx sdk.Context, portID, channelID string) (icatypes.Metadata, error) {
+	appVersion, found := k.GetAppVersion(ctx, portID, channelID)
+	if !found {
+		return icatypes.Metadata{}, errorsmod.Wrapf(ibcerrors.ErrNotFound, "app version not found for port %s and channel %s", portID, channelID)
+	}
+
+	var metadata icatypes.Metadata
+	if err := icatypes.ModuleCdc.UnmarshalJSON([]byte(appVersion), &metadata); err != nil {
+		// UnmarshalJSON errors are indeterminate and therefore are not wrapped and included in failed acks
+		return icatypes.Metadata{}, errorsmod.Wrapf(icatypes.ErrUnknownDataType, "cannot unmarshal ICS-27 interchain accounts metadata")
+	}
```

Comment:
I guess one slight consideration here is that the error code changes depending on whether you wrap the error or not (the error code is included in the acknowledgement). I don't have a preference one way or another, but given that we do have a strong guarantee that we will wipe non-deterministic info from an error before creating a error acknowledgement, I could be okay with wrapping the error

Reply:
I haven't wrapped them since this is what we agreed in one of the calls. If you don't have a preference, I'm happy to keep it as is.